pkg/job/local_job.go

```diff
@@ -0,0 +1,83 @@
+package job
+
+import (
+	"bytes"
+	"context"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/openshift/assisted-service/internal/common"
+	"github.com/openshift/assisted-service/internal/events"
+	"github.com/openshift/assisted-service/pkg/generator"
+	logutil "github.com/openshift/assisted-service/pkg/log"
+	"github.com/sirupsen/logrus"
+)
+
+type LocalJob interface {
+	Execute(pythonCommand string, pythonFilePath string, envVars []string, log logrus.FieldLogger) error
+	generator.ISOInstallConfigGenerator
+}
+
+type localJob struct {
+	Config
+	log logrus.FieldLogger
+}
+
+func NewLocalJob(log logrus.FieldLogger, cfg Config) *localJob {
+	return &localJob{
+		Config: cfg,
+		log:    log,
+	}
+}
+
+func (j *localJob) Execute(pythonCommand string, pythonFilePath string, envVars []string, log logrus.FieldLogger) error {
+	cmd := exec.Command(pythonCommand, pythonFilePath)
+	cmd.Env = envVars
+	var out bytes.Buffer
+	cmd.Stdout = &out
+	if err := cmd.Run(); err != nil {
+		log.Infoln(pythonFilePath + " executed with error: " + err.Error())
```

Comment:
why not log.WithError? and why is the log level info?

Reply:
I think it was Info to diagnosis an issue with the unit test. I've switched it back to Error. Thanks.